src/main/java/com/example/demo/controller/CategoryController.java

```diff
@@ -0,0 +1,75 @@
+package com.example.demo.controller;
+
+import com.example.demo.dto.book.BookDtoWithoutCategoryIds;
+import com.example.demo.dto.category.CategoryDto;
+import com.example.demo.service.book.BookService;
+import com.example.demo.service.category.CategoryService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "Category management", description = "The Category API")
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+    private final CategoryService categoryService;
+    private final BookService bookService;
+
+    @PreAuthorize("hasRole('ROLE_ADMIN')")
+    @Operation(summary = "Create a category", description = "Create a new category")
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public CategoryDto createCategory(@RequestBody @Valid CategoryDto requestDto) {
+        return categoryService.createCategory(requestDto);
+    }
+
+    @PreAuthorize("hasRole('ROLE_USER')")
```

Comment:
It's not necessary to check everywhere for user role,  cause this endpoints is only for authenticated users